### PR TITLE
frontend - add redux and set api requests as actions

### PR DESCRIPTION
**Pull request by otomer**
added the following folders structure to the project:
- actions\     
- reducers\
- store\

1. moved inline api requests from components to actions 
2. retrieving data via reducers
3. init store on app component

### CODE REVIEW

File: frontend/src/actions/cuisines.actions.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import axios from 'axios';
+
+export const API_URL_GET_CUISINES = 'cuisines';
+export const REQUEST_CUISINES = 'REQUEST_CUISINES';
+export const RETRIEVED_CUISINES = 'RETRIEVED_CUISINES';
+
+export const requestCuisines = () => ({ type: REQUEST_CUISINES });
```
Comment (GeReV) [inline frontend/src/actions/cuisines.actions.js:7]: I highly recommend [`redux-actions`](https://redux-actions.js.org).

File: frontend/src/components/CuisinesNavigation/index.js [hunk 2]
```diff
@@ -3,25 +3,26 @@ import PropTypes from 'prop-types';
...
+            ))}
         </ul>
       </div>
     </nav>
   );
 };
 
 CuisinesNavigation.propTypes = {
-  items: PropTypes.array.isRequired,
+  cuisines: PropTypes.array,
```
Comment (GeReV) [inline frontend/src/components/CuisinesNavigation/index.js:25]: Could be useful to use `PropTypes.arrayOf(...)`.

File: frontend/src/containers/CategoriesContainer/index.js [hunk 3]
```diff
@@ -1,21 +1,39 @@
...
+  return {
+    isFetching,
+    lastUpdated,
+    cuisines,
+  };
+};
+
+function mapDispatchToProps(dispatch) {
+  return { fetchCuisines: bindActionCreators(fetchCuisines, dispatch) };
+}
```
Comment (GeReV) [inline frontend/src/containers/CategoriesContainer/index.js:34]: In this case, you can write it in shorthand:
```js
const mapDispatchToProps = {
  fetchCuisines,
};
```

File: frontend/src/reducers/cuisines.reducer.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
+import { REQUEST_CUISINES, RETRIEVED_CUISINES } from '../actions';
+
+const cuisinesReducer = (
```
Comment (GeReV) [inline frontend/src/reducers/cuisines.reducer.js:3]: `redux-actions` offers a really nice helper called `handleActions` for building reducers.

**Review by GeReV — APPROVED** [review]

File: frontend/src/components/CuisinesNavigation/index.js (hunk 2 above)
Comment (otomer) [reply frontend/src/components/CuisinesNavigation/index.js:25]: Strongly agree

File: frontend/src/containers/CategoriesContainer/index.js (hunk 3 above)
Comment (otomer) [reply frontend/src/containers/CategoriesContainer/index.js:34]: Hey that's a nice one! great suggestion

File: frontend/src/actions/cuisines.actions.js (hunk 1 above)
Comment (otomer) [reply frontend/src/actions/cuisines.actions.js:7]: looks interesting, bit simpler than the classic way of redux, definitely reviewing it

**Pull request merged**